### ISSUE TITLE
support relative URIs per RFC1808 and RFC3986

### ISSUE THREAD

**Issue by coderextreme** [mafintosh/is-my-json-valid#105]
Need to support leading . and ..

**Comment by coderextreme** [comment]
or at least leading ..

**Comment by coderextreme** [comment]
[CC.txt](https://github.com/mafintosh/is-my-json-valid/files/108430/CC.txt)
[Schema.txt](https://github.com/mafintosh/is-my-json-valid/files/108431/Schema.txt)

**Comment by emilbayes** [comment]
@coderextreme Could you elaborate on your request? I don't see any use of relative paths in your examples. Is this in regards to remote `$ref`s? They're not supported by this module

**Comment by coderextreme** [comment]
No, it's not remote refs, It's properties (@url, @*Url) marked as "format": "uri" in my schema.  I need to validate URI's in my documents, and documents with a URI of .. don't pass.  I have overridden the URI format for now.  Thanks!

**Comment by emilbayes** [comment]
The following works fine for me:

```js
var validate = require('is-my-json-valid')({type: 'string', format: 'uri'})

validate('http://hello.world/../hello.txt') // true
```

**Comment by handrews** [comment]
@emilbayes @coderextreme format "uri" was poorly specified in draft-04.  Technially, it indicates a full URI (with scheme).  The "uri-reference" format added in draft-06 is what would handle `"../hello.txt"` as a URI reference to be resolved against the current base per RFC 3986
